### ISSUE TITLE
Support for rstrnt-... comands on DUT provisioned outside of beaker

### ISSUE THREAD

**Issue by BeeGrech** [teemtee/tmt#1027]
all kernel tests use restraintd commands like rstrnt-report-result for example. These commands are not recognized on systems provisioned outside of beaker. 
In some cases the command can be called many times per one test. 

Expected behavior: TMT would update the test result

Observed behavior: TMT Captures the output marking a test as failed or skipped, but TMT report shows test as passed.

Log Output:

A report from tmt with this may look like something similar:
15:50:06     report
15:50:06         workdir: /var/tmp/tmt/run-045/rt_cert_test/report
15:50:06         how: display
15:50:06         order: 50
15:50:06             pass /rt_cert_test
15:50:06                 output.txt: /var/tmp/tmt/run-045/rt_cert_test/execute/data/rt_cert_test/output.txt
15:50:06             Read file '/var/tmp/tmt/run-045/rt_cert_test/execute/data/rt_cert_test/output.txt'.
15:50:06                 content: 
                    Warning: Permanently added '[127.0.0.1]:10022' (ED25519) to the list of known hosts.
                    This test runs/is relevant only on Red Hat Enterprise Linux.
                    runtest.sh: line 52: rstrnt-report-result: command not found
15:50:06         summary: 1 test passed

So the test is marked as passed, but the output does not appear to match. In some cases this could represent a false pass. In this case the test would have reported "SKIP" to restraintd, in some cases it can be FAIL

--
At this time adding 3 restraint commands should be sufficient for adapting tests-
rstrnt-report-result
rstrnt-report-log
rstrnt-abort

**Comment by lukaszachy** [comment]
FTR: There are other commands as well: https://restraint.readthedocs.io/en/latest/commands.html#restraintd-intro
And plenty of legacy  (which are likely used too) https://restraint.readthedocs.io/en/latest/remove_rhts.html#legacy-rhts-cmds

If we do this we might at least to the log submit too (even though it will be simple copy to TMT_TEST_DATA dir).

**Comment by psss** [comment]
Thanks for reporting this. I believe the implementation of `rstrnt-report-result` which would store test result and optionally submit log or custrom metric should be fairly easy. But as @lukaszachy mentioned there can be much more command and we should consider what would be the best approach.

I would say, if there is a reasonable number of frequently used `restraint` commands it should be ok to create backward compatible alternatives in `tmt`. However, if we find out that many tests are using many of these commands it would probably make more sense to go with the original idea #197 to implement a `restraint` plugin for the `execute` step.

**Comment by rasibley** [comment]
hi @psss @thrix just an fyi this is needed for Automotive testing as we plan to use Testing Farm + TMT for our execution, we will be attempting to work around this for kernel by possibly running restraint kernel tests in CKI's environment for now, if there is some way to prioritize this work that would be nice as this is needed for RHIVOS certification. @BeeGrech would you mind changing the summary to capture full restraint support not just restraint-report-result ? Thank you !

**Comment by lukaszachy** [comment]
@rasibley Could you say which other commands are necessary?

If it is about full support than this should be closed as duplicate of #197 (which should get the priority).
Reason is that many of restraint commands are out of scope what tmt currently can do (e.g. rstrnt-adjust-watchdog)

**Comment by thrix** [comment]
I agree, let's close #197 as the duplicate and track the implementation here.

I got information that @pfdaly should start work on this soon.

@pfdaly feel free to contact @lukaszachy, @psss (he is out this week) or me to discuss details if you want. I will also add you to our development meetings on Tuesday.

@psss had made a nice code intro video here: https://issues.redhat.com/browse/TT-167

**Comment by BeeGrech** [comment]
also adding rstrnt-abort
those are the top 3 for kernel tests

**Comment by thrix** [comment]
CKI uses these restraint functions:

```
rstrnt-abort
rstrnt-backup
rstrnt-reboot
rstrnt-report-log
rstrnt-report-result
rstrnt-restore
rstrnt-sync
rstrnt-sync-block
rstrnt-sync-set
```

**Comment by thrix** [comment]
All restraint commands: https://restraint.readthedocs.io/en/latest/commands.html

**Comment by pfdaly** [comment]
I was presented with three options for this ticket:

1) To convert the existing CKI tests to TMT format.
2) To create a Restraint plugin in TMT.
3) To create backward compatible alternatives in TMT.

Having reviewed the work involved in each, and having discussed the topic, it has been agreed that I shall proceed with the implementation of option (3).
I will begin working on this now.

**Comment by happz** [comment]
Based on https://github.com/teemtee/tmt/issues/1027#issuecomment-1084478173, there do not seem to be much left to discuss, and the issue can be closed now. Feel free to file a new one should you run into the need for additional commands or discussions.
